syntax = "proto3";

package message;

service MessageService {
  rpc AssignAB (ABRequest) returns (ABResponse) {}
  rpc Factorial (FactorialRequest) returns (FactorialResponse) {}
  rpc MostFrequentLetter (TextRequest) returns (LetterResponse) {}
}

message ABRequest {
  string user_id = 1;
}

message ABResponse {
  string group = 1; // "A" or "B"
}

message FactorialRequest {
  int32 n = 1;
}

message FactorialResponse {
  string value = 1; // big numbers as string
}

message TextRequest {
  string text = 1;
}

message LetterResponse {
  string letter = 1;
}

